addons/controllers/cpi/vspherecpiconfig_utils.go

```diff
@@ -95,15 +97,13 @@ func (r *VSphereCPIConfigReconciler) mapCPIConfigToDataValuesNonParavirtual( //
 	d.TLSThumbprint = vsphereCluster.Spec.Thumbprint
 	d.Server = vsphereCluster.Spec.Server
 
-	// derive vSphere username and password from the <cluster name> secret
-	clusterSecret, err := r.getSecret(ctx, cluster.Namespace, cluster.Name)
+	// derive vSphere username and password using CAPV util function
+	clusterCredentials, err := capvidentity.GetCredentials(ctx, r.Client, vsphereCluster, capvmanager.DefaultPodNamespace)
```

Comment:
Can we expect that this ```DefaultPodNamespace``` always get populated correctly?

Reply:
no, the DefaultPodNamespace is always `capv-system`, we can't gurantee its existence. if it doesn't exist, the `GetCredentials` will just use the cluster's namespace